File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        count = {}
        for num in nums1:
            count[num] = count.get(num,0)+1
        
        temp = []
        for x in nums2:
            if x in count and count[x]>0:
                count[x] = count.get(x,0)-1
                temp.append(x)
                
        return temp

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    let res = nums1
    let tmp = nums2
    
    let final = []
    
    if(nums1.length<nums2.length){
        res = nums2
        tmp = nums1
    }
    
    for(let i=0; i<tmp.length ; i++){
        let cur = tmp[i]
        if(res.includes(cur)){
            final.push(cur)
            res.splice(res.indexOf(cur), 1)
        }
    }
    
    return final
};